### PR TITLE
Migrate Extension to React and Typescript

### LINKED ISSUE #7: Locally define key
i keep accidentally uploading the key, find a way to read key from local file without exposing to public

(note, the key in commits no longer works)

### LINKED ISSUE #13: Incorporate local storage for default options
Going to need to make a config file for this. Probably json just to keep in line with other data files.

### LINKED ISSUE #14: Nuke all forEach()
forEach adds unnecessary brackets, negatively impacting code readability and hurts my soul

### LINKED ISSUE #15: Move all global consts into some other file
Not sure if we want to store this in a js file or a json file or something.

### PR DESCRIPTION
**Pull request by quinnha**
Migrate Extension to React and Typescript

## Description
Basically just created a react app and loaded existing files in it. translated content.js to ts. also added some boilerplate for for william to use to connect the front end with the backend

Need to refactor in UI changes + folders into react file structure, later pr

## Other
i hate git, i miss critique

### CODE REVIEW

File: package-lock.json
Comment (wastadtlander) [inline package-lock.json:1]: I was wrong btw we don't gitignore this file.

File: public/logo192.png
Comment (wastadtlander) [inline public/logo192.png:1]: Guessing we need to replace this with a logo eventually?

File: public/index.html
Comment (wastadtlander) [inline public/index.html:1]: So this is equivalent to what will be popup.html right?

File: public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "Chrome React SEO Extension",
```
Comment (wastadtlander) [inline public/manifest.json]: Chagne name.

File: public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "The power of React and TypeScript for building interactive Chrome extensions",
```
Comment (wastadtlander) [inline public/manifest.json]: Change description.

File: public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "version": "1.0",
+  "manifest_version": 3,
+  "action": {
+    "default_popup": "index.html",
+    "default_title": "Open the popup"
```
Comment (wastadtlander) [inline public/manifest.json]: Change title to like `Comment Translator`

File: public/manifest.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  "icons": {
```
Comment (wastadtlander) [inline public/manifest.json]: We can probably dump the icons for now.

File: public/manifest.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    "48": "logo192.png",
+    "128": "logo192.png"
+  },
+  "permissions": [
+    "activeTab"
+ ],
+ "content_scripts": [
+  {
+    "matches": ["https://www.youtube.com/watch?v=*"],
+      "js": ["./static/js/content.js"]
```
Comment (wastadtlander) [inline public/manifest.json:14]: Line up and fix the alignment of the bracket and brace below.

File: public/robots.txt
Comment (wastadtlander) [inline public/robots.txt:1]: What is this for?

File: src/App.test.tsx
Comment (wastadtlander) [inline src/App.test.tsx:1]: Can we dump this test for now just to cleanup what we don't need.

File: src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        if (response && response.response && response.response.divIds) {
+          setDivIds(response.response.divIds);
+        } else {
+          console.error("Invalid response structure:", response);
+        }
+      });
+    });
+  };
+
+  const getUILanguage = () => {
```
Comment (wastadtlander) [inline src/App.tsx:33]: Wait so is this where the UI stuff would go?

File: src/chromeServices/DOMEvaluator.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+  // Prepare the response object
+  const response: DOMMessageResponse = {
+    success: true, // Indicate successful processing
+  };
+
+  sendResponse(response);
+};
+
+/**
```
Comment (wastadtlander) [inline src/chromeServices/DOMEvaluator.ts]: one line

File: src/chromeServices/content.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
+/** Google Translate API */
+import { TranslateAPIResponse } from "../types";
+
+const key = process.env.REACT_APP_GOOGLE_TRANSLATE_API_KEY; // Insert Key Here
+const translateURL = `https://translation.googleapis.com/language/translate/v2?key=${key}`;
+const commentWrapper = document.body;
+
+/**
+ * Observer to translate new comments on load
```
Comment (wastadtlander) [inline src/chromeServices/content.ts]: one line

File: src/index.tsx
Comment (wastadtlander) [inline src/index.tsx:1]: ok i see this is the parent tsx file that returns html.

File: src/reportWebVitals.ts
Comment (wastadtlander) [inline src/reportWebVitals.ts:1]: whats this?

**Review by wastadtlander — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (quinnha) [reply package-lock.json:1]: swag

File: public/index.html
Comment (quinnha) [reply public/index.html:1]: no, app.tsx will be whats equivalent to popup.html, index.html just mounts the react component - can also link to static assets like icons, fonts, and any external CSS files that are not part of the React component structure

File: public/logo192.png
Comment (quinnha) [reply public/logo192.png:1]: yea i just left it in bc its default, can do whenever we figure out logos and brandings and stuff

File: public/manifest.json (hunk 1 above)
Comment (quinnha) [reply public/manifest.json]: woopsies

File: public/manifest.json (hunk 2 above)
Comment (quinnha) [reply public/manifest.json]: woopsies (2)

File: public/manifest.json (hunk 4 above)
Comment (quinnha) [reply public/manifest.json]: was just placeholder but can remove

File: public/robots.txt
Comment (quinnha) [reply public/robots.txt:1]: autogend, its for the webcrawlers, and lets people google our extension

File: src/App.test.tsx
Comment (quinnha) [reply src/App.test.tsx:1]: sure, guess we can just add it back later when we figure out testing

File: src/App.tsx (hunk 6 above)
Comment (quinnha) [reply src/App.tsx:33]: in the return statement slightly below is where the UI is rendered, these are just functions -> this one in particular just gets the language of the browser

File: src/index.tsx
Comment (quinnha) [reply src/index.tsx:1]: yea this is where it be rendering

File: src/reportWebVitals.ts
Comment (quinnha) [reply src/reportWebVitals.ts:1]: see index.tsx

```
// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```

**Review by quinnha — COMMENTED** [review]
done

**Review by wastadtlander — APPROVED** [review]

**Pull request merged**